File: 백준/Bronze/2798. 블랙잭/블랙잭.js
let [[_, min], input] = require("fs")
  .readFileSync("/dev/stdin")
  .toString()
  .trim()
  .split("\n")
  .map((i) => i.split(" "))
  .map((i) => i.map(Number));
let max = 0;

for (let i = 0; i < input.length; i++) {
  for (let j = i + 1; j < input.length; j++) {
    for (let n = j + 1; n < input.length; n++) {
      const item = input[i] + input[j] + input[n];
      if (item > max && item <= min) {
        max = item;
      }
    }
  }
}
console.log(max);

File: 백준/Bronze/2798. 블랙잭/블랙잭.cc
#include <iostream>
using namespace std;

int main() {
    int n,m,c[100],max=0;
    cin >> n >> m;
    for(int i=0; i<n; i++) {
        cin >> c[i];
    };

    for(int i=0; i<n; i++) {
        for(int j=i+1; j<n; j++) {
            for(int k=j+1; k<n; k++) {
                if(c[i]+c[j]+c[k] > max && c[i]+c[j]+c[k] <= m) {
                    max=c[i]+c[j]+c[k];
                }
            }
        }
    }

    cout << max << "\n";

    return 0;
}